client/components/isomorphic-id/index.js

```diff
@@ -0,0 +1,42 @@
+/** @ssr-ready **/
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { incrementIsomorphicId } from 'state/ui/actions';
+import { getCurrentIsomorphicId } from 'state/ui/selectors';
+
+export default function addUniqueComponentId( Component ) {
+	class ComponentWithId extends React.Component {
+		constructor( props ) {
+			super( props );
+
+			this.componentId = this.props.currentIsomorphicId;
+			this.props.incrementIsomorphicId();
```

Comment:
Right. It's also important that `incrementIsomorphicId()` is synchronous so it has to be a pure Redux action creator, not a thunk.